### PR TITLE
Resource sharing plugin2

### PR DESCRIPTION
**Pull request by elpaso**
This is the QGIS3 port of the Resource Sharing Plugin, developed by @akbargumbira during 2016 GSOC, merged as a core Python plugin.

QEP: https://github.com/qgis/QGIS-Enhancement-Proposals/issues/58
Plugin: http://plugins.qgis.org/plugins/qgis_resource_sharing/
Previous discussions: 
- https://github.com/qgis/QGIS/pull/4087
- https://lists.osgeo.org/pipermail/qgis-developer/2017-January/046501.html

### CODE REVIEW

**Comment by akbargumbira** [comment]
Hi Ale,

I apologize for the unresponsiveness around this time. But I'll be free
after tomorrow for a few days and will try this PR.

Cheers

**Comment by elpaso** [comment]
@akbargumbira thanks. It looks like the fedora users have some problems with the webkit widgets as @nyalldawson  mentioned in the other PR. Can you have a look?

**Comment by nyalldawson** [comment]
@elpaso FYI - i'm mid-way through a review on this. It's a WIP but just wanted to give you a heads up that i'm working on it!

**Comment by akbargumbira** [comment]
Hey @nyalldawson,  I am on 'school' holiday too for a few days. Let me know if you want me to do something particular. I don't have machine with fedora though, I can't replicate your problem

File: python/plugins/ResourceSharing/test/data/collections/test_collection/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (nyalldawson) [inline python/plugins/ResourceSharing/test/data/collections/test_collection/LICENSE.txt]: Does this license apply to the pokemon images too? I'm a bit concerned we might run into trouble if we distribute them

File: python/plugins/ResourceSharing/test/test_symbol_xml_extractor.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import sys
+
+# append plugin folder to path
+sys.path.append(os.path.join(os.path.dirname(__file__), os.path.pardir))
+
+# Note that the instance returned in the import test is
+# QGIS 2.x: QgsRandomColorRamp
+# QGIS 3.x is QgsLimitedRandomColorRamp
+
+try:  # QGIS 2.x
```
Comment (nyalldawson) [inline python/plugins/ResourceSharing/test/test_symbol_xml_extractor.py]: this 2.x compatibility code should be removed

File: python/plugins/ResourceSharing/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+FOREACH(dir
+processing
+resource_sharing
+resource_sharing/gui
+resource_sharing/gui/ui
+resource_sharing/repository_handler
+resource_sharing/resource_handler
+resources
+resources/js
```
Comment (nyalldawson) [inline python/plugins/ResourceSharing/CMakeLists.txt]: is this outdated? i can't see js/boostrap/etc in this PR

**Review by nyalldawson — COMMENTED** [review]
I think you're missing an ADD_SUBDIRECTORY(ResourceSharing) in python/plugins/CmakeLists.txt 

Code looks good to me! (with some notes above about removal of old 2.x compatibility code)

Before this gets merged can I get confirmation that the plugin will be removed from the plugin repo for 3.0? I really don't want to see the same situation which occurred with processing where versions were included both in the main installer & via the plugin repo, which resulted in users getting stuck on old versions of the plugin unless they manually cleaned out their plugin folder. We should be able to avoid this this time as everyone will start with a clean config folder with no existing version of the plugin, but we also need to make sure that they cannot install the plugin from the repo too.

**Comment by nyalldawson** [comment]
(also apologies for the mega slow review...)

**Comment by elpaso** [comment]
@nyalldawson thanks for reviewing, I've started now to implement the required changes.

File: python/plugins/ResourceSharing/resource_sharing/gui/resource_sharing_dialog.py [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+
+    def reject(self):
+        """Slot when the dialog is closed."""
+        # Serialize collections to settings
+        self.repository_manager.serialize_repositories()
+        self.done(0)
+
+    def open_help(self):
+        """Open help."""
+        doc_url = QUrl('http://www.akbargumbira.com/qgis_resources_sharing')
```
Comment (elpaso) [inline python/plugins/ResourceSharing/resource_sharing/gui/resource_sharing_dialog.py]: @alexbruy , @SrNetoChan can you help here?

File: python/plugins/ResourceSharing/test/data/collections/test_collection/LICENSE.txt (hunk 1 above)
Comment (elpaso) [inline python/plugins/ResourceSharing/test/data/collections/test_collection/LICENSE.txt]: no clue here, I guess we should just replace the images

Comment (elpaso) [inline python/plugins/ResourceSharing/test/data/collections/test_collection/LICENSE.txt]: @pcav can you take help with this? ^

**Comment by akbargumbira** [comment]
@elpaso Can I have push access to your fork so that I can commit there?

**Comment by elpaso** [comment]
@akbargumbira you should be good to go, the Travis tests are failing but I couldn't find a reason for the failure. TODO is the integration of the help pages into the main QGIS manual, but I guess this can also be done after the merge.

**Comment by pcav** [comment]
@akbargumbira have you already done it, or need my help?
Cheers.

**Comment by elpaso** [comment]
Lost motivation.

**Comment by nyalldawson** [comment]
> And QtWebKit is dying, so similarly with the frontend.

That's not true anymore... Upstream changed their mind, it got resurrected and won't be culled anymore.

**Comment by borysiasty** [comment]
> That's not true anymore... Upstream changed their mind, it got resurrected and won't be culled anymore.

Ah, so it seems I misunderstood the recent discussion on bugs.debian.org - they were only mentioning Qt4.QtWebkit removal. Thanks for correction.

Anyway, the Manager is screaming for a facelift. I started porting it to QtQuick a few years ago, but it was too early for some of the supported OSes. Next year I plan to give it another try.

**Pull request closed without merging**